From e7c9fda65257214e7bb9524bb142a76d6e7c128b Mon Sep 17 00:00:00 2001
From: "sj.cha" <sj.cha@lge.com>
Date: Fri, 3 Mar 2017 06:54:59 +0900
Subject: [PATCH 7/9] servicemanager: Restart keystore and gatekeeper after
 crash

Reproduce step :
- Kill servicemanager process

frequency : 100%

Reason of issue :
As you know, servicemanager manages almost framework services.
However when servicemanager killed by unknown reason, android restart servicemanager.
In this case, keystore and gatekeeper deamon should be restart.
But current android source code doesn't consider those case.
So, I want to disscuss about this changes with you.
(It is able to so fixed in theirs(keystore.rc...) rc files.
But I'm not sure what is the best changes.)

Solution : Adding restart services syntax from rc file

(cherry picked from commit b2dead1d1d254ecfe33b6948c9cbc606b05b99a0)

Signed-off-by: SangJin Cha <sj.cha@lge.com>
Change-Id: Iab02b54417406dd84a1e970dc1ded6437183d0c0
Signed-off-by: mydongistiny <jaysonedson@gmail.com>
---
 cmds/servicemanager/servicemanager.rc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmds/servicemanager/servicemanager.rc b/cmds/servicemanager/servicemanager.rc
index d336a436d..4d93cb4c7 100644
--- a/cmds/servicemanager/servicemanager.rc
+++ b/cmds/servicemanager/servicemanager.rc
@@ -11,5 +11,7 @@ service servicemanager /system/bin/servicemanager
     onrestart restart inputflinger
     onrestart restart drm
     onrestart restart cameraserver
+    onrestart restart keystore
+    onrestart restart gatekeeperd
     writepid /dev/cpuset/system-background/tasks
     shutdown critical
-- 
2.14.1

